torchao/dtypes/nf4tensor.py

```diff
@@ -853,7 +865,13 @@ def fsdp_post_all_gather(
             return
 
         return nf4_constructor(
-            tensor_meta,
+            # tensor_meta,
+            tensor_meta.original_shape,
+            tensor_meta.original_strides,
+            tensor_meta.storage_offset,
+            tensor_meta.dtype,
+            tensor_meta.device,
+            tensor_meta.requires_grad,
```

Comment:
Without this change, when tracing through this function it will complain that `tensor_meta` (which is of `SubclassTensorArgs` type, a UserDefinedObjectVariable in Dynamo) cannot be proxyed 🤔

Reply:
I filed an issue for this one with more details here: https://github.com/pytorch/pytorch/issues/133858. I think this... might be fixable if we beef up dynamo to not treat `SubclassTensorArgs` as a `UserDefinedObjectVariable` like you said. A workaround would be to make NF4Tensor use a `NamedTuple` instead of a `dataclass` here, although it would be nice to just fix this properly.